docs/DeveloperGuide.md

```diff
@@ -1,38 +1,360 @@
+* It saves the name of the user for future uses.
+* It ends the program when called to.
+
+Modganiser consists of :
+
+1. `Storage`
+2. `Parser`
+3. `ModuleList`
+4. `Print`
+
+**How the components interact with each other**
+
+When using the command `add /CS2113T /4 /CORE /1 /1` ,
+The Sequence Diagram shows how the components interact with each other:
+
+{image}
+
+### 1. **Storage Component**
+
+Firstly, the {explain workflow}
+
+{uses}
+*  It can save the name of the user into a file.
+* The program can load data from the file when Modganiser starts up again.
+* It can save changes to the file after each change such as an addition or deletion.
+  This is done by using the methods
+
+
+### 2. **Parser Component**
+
+Firstly, the {explain workflow}
+
+{uses}
+* It calls the relevant methods according to the user input
+
+
+### 3. **ModuleList Component**
+
+Firstly, the {explain workflow}
+
+{uses}
+*
+
+### 4. **Print Component**
+
+
+Firstly, the {explain workflow}
+
+{uses}
+*
+
+## 3. Design and Implementation
+
+---
+
+### 3.1. ManPage 
+
+The ManPage is a class that is used to display parts of the manual page that is requested by the user.
+It has multiple methods for separate features.
+
+**Implementation**
+
+The sequence by which the ManPage class handles the `man` command is as follows:
+1) The `Parser` class extracts the other fields of the user input, and calls upon the `checkUserInput()` method
+   which calls upon the `checkManPageInput()`.
+
+2) The method then uses the methods in the ManPage class to print out the requested feature.
+
+`ManPage#printManPage()` - Prints the Command Summary.
+
+`ManPage#printManPageAdd()` - Prints description of how to use the add feature.
+
+`ManPage#printManPageList()` - Prints description of how to use the list feature.
+
+`ManPage#printManPageEdit()` - Prints description of how to use the edit feature.
+
+`ManPage#printManPageFind()` - Prints description of how to use the find feature.
+
+`ManPage#printManPageDelete()` - Prints description of how to use the delete feature.
+
+`ManPage#printManPageCalculator()` - Prints description of how to use the grade calculator.
+
+`ManPage#printManPageUpdateGrade()` - Prints description of how to input grades.
+
+`ManPage#printManPageExit()` - Prints description of how to use the exit feature.
+
+![ManPage](diagrams/ManPage.png)
+
+**Design Considerations**
+
+**Alternative 1** : Prints the manual page individually as features. 
+Pros: Avoids wordy and long manual page when command is man, ability to have more description
+for specific features. Avoids long methods. 
+Cons: Extra step of typing another command to understand a feature well.
+
+**Alternative 2**: Combining all feature descriptions in a single manual page. 
+Pros: Avoid extra step of typing another command to understand a feature well.
+Cons: Wordy and long manual page when command is man, user has to scroll through
```

Comment:
![image](https://user-images.githubusercontent.com/1620654/228721988-93e04f1e-af7f-4636-8aa2-b0f3733c321f.png)

Similarly the pros and cons aren't on new lines, and perhaps the command is man should highlight that "man" as a code input